code/modules/food_and_drinks/food/foods/pizza.dm

```diff
@@ -447,100 +446,140 @@
+#define PIZZA_BOMB_PRIMED	 2
+#define PIZZA_BOMB_DISARMED  3
+
 //////////////////////////
 //		Pizza bombs		//
 //////////////////////////
 /obj/item/pizzabox/pizza_bomb
 	/// Adjustable timer
 	var/timer = 1 SECONDS
-	var/timer_set = FALSE
-	var/primed = FALSE
-	var/disarmed = FALSE
+	var/pizza_bomb_status = PIZZA_BOMB_NOT_ARMED
 	var/wires = list("orange", "green", "blue", "yellow", "aqua", "purple")
 	var/correct_wire
-	var/armer //Used for admin purposes
-	var/opener //Ditto
+	var/mob/armer //Used for admin purposes
+	var/mob/opener //Ditto
 
-/obj/item/pizzabox/pizza_bomb/update_desc()
-	if(primed)
-		return
+/obj/item/pizzabox/pizza_bomb/Initialize(mapload)
+	correct_wire = pick(wires)
+	var/obj/item/pizzabox/mimic_box = pick(subtypesof(/obj/item/pizzabox) - typesof(/obj/item/pizzabox/pizza_bomb))
+	box_tag = mimic_box.box_tag
 	return ..()
 
-/obj/item/pizzabox/pizza_bomb/update_icon_state()
-	if(primed)
-		return
+/obj/item/pizzabox/pizza_bomb/Destroy()
+	armer = null
+	opener = null
 	return ..()
 
+/obj/item/pizzabox/pizza_bomb/update_name()
+	. = ..()
+	if(!open)
+		name = "pizza box"
+		return
+	name = "pizza bomb"
+
+/obj/item/pizzabox/pizza_bomb/update_desc()
+	if(!open)
+		return ..()
+
+	switch(pizza_bomb_status)
+		if(PIZZA_BOMB_DISARMED)
+			desc = "A devious contraption, made of a small explosive payload hooked up to pressure-sensitive wires. It's disarmed."
+		if(PIZZA_BOMB_PRIMED)
+			desc = "OH GOD THAT'S NOT A PIZZA"
+		else
+			desc = "It seems inactive."
+
+/obj/item/pizzabox/pizza_bomb/update_icon_state()
+	if(!open)
+		return ..()
+
+	switch(pizza_bomb_status)
+		if(PIZZA_BOMB_DISARMED)
+			icon_state = "pizzabox_bomb_[correct_wire]"
+		if(PIZZA_BOMB_PRIMED)
+			icon_state = "pizzabox_bomb_active"
+		else
+			icon_state = "pizzabox_bomb"
+
 /obj/item/pizzabox/pizza_bomb/AltClick(mob/user)
 	attack_self(user)
 
 /obj/item/pizzabox/pizza_bomb/attack_self(mob/user)
-	if(disarmed)
-		to_chat(user, "<span class='notice'>[src] is disarmed.</span>")
-		return
-	if(!timer_set)
-		name = "pizza bomb"
-		desc = "It seems inactive."
-		icon_state = "pizzabox_bomb"
-		timer_set = TRUE
+	if(pizza_bomb_status == PIZZA_BOMB_NOT_ARMED)
+		open = TRUE
+		update_appearance(UPDATE_NAME|UPDATE_DESC|UPDATE_ICON)
+
 		var/new_timer = tgui_input_number(user, "Set a timer, from one second to ten seconds.", "Timer", timer / 10, 10, 1)
-		if(isnull(new_timer))
-			return
-		if(!in_range(src, user) || issilicon(user) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) || user.restrained())
-			timer_set = FALSE
-			name = "pizza box"
-			desc = "A box suited for pizzas."
-			icon_state = "pizzabox1"
+		if(isnull(new_timer) || !in_range(src, user) || issilicon(user) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) || user.restrained())
+			open = FALSE
+			update_appearance(UPDATE_NAME|UPDATE_DESC|UPDATE_ICON)
 			return
+
 		timer = new_timer SECONDS
-		icon_state = "pizzabox1"
+		pizza_bomb_status = PIZZA_BOMB_TIMER_SET
+		armer = user
```

Comment:
if the player who armed the bomb leaves the body of the mob that performed the action do we lose the ckey in the logs?

Reply:
Might be better to hang onto their mind, maybe

Reply:
![image](https://github.com/user-attachments/assets/57a9c332-e6df-41cc-9f19-f30c5ef408e2)